library/Haste/Image/Image.php

```diff
@@ -202,4 +202,252 @@ public static function addWatermark($image, $watermark, $position = 'br', $targe
+     *
+     * See findImages() for more option keys.
+     *
+     * @param array $options
+     *
+     * @return string
+     */
+    public static function getGallery(array $options)
+    {
+        $images = static::findImages($options);
+
+        if (isset($options['sortBy'])) {
+            static::sortImages($images, $options['sortBy'], $options);
+        }
+
+        $body = [];
+
+        foreach ($images as $k => $image) {
+            $stdClass = $image['templateData'];
+            $stdClass->class = $k % 2 === 0 ? 'odd' : 'even';
+            $body['image'][] = $stdClass;
+        }
+
+        $template = new \FrontendTemplate($options['template'] ?: 'gallery_default');
+        $template->body = $body;
+        return $template->parse();
+    }
+
+    /**
+     * Prepares the data for a typical Contao gallery fetching image data, meta data,
+     * responsive images data etc. based on options.
+     *
+     * Possible options keys:
+     *
+     *  - multiSRC (mutliple UUIDS, either as a serialized string or already as array)
+     *  - singleSRC (one UUID)
+     *  - sortBy (a key specifying the sort options. See sortImages() for options and requirements (some sorting options require more info)
+     *
+     * See prepareImage() for more option keys.
+     *
+     * @param array $options
+     *
+     * return array
+     */
+    public static function findImages(array $options)
+    {
+        $images = [];
+
+        if (isset($options['multiSRC'])) {
+            $files = deserialize($options['multiSRC'], true);
+        } elseif (isset($options['singleSRC'])) {
+            $files = [$options['singleSRC']];
+        } else {
+            throw new \InvalidArgumentException('Need to provide either multiSRC or singleSRC.');
+        }
+
+        $fileModels = \FilesModel::findMultipleByUuids($files);
+
+        if (null === $fileModels) {
+            return [];
+        }
+
+        foreach ($fileModels as $fileModel) {
+            // Single files
+            if ('file' === $fileModel->type) {
+                $file = new \File($fileModel->path, true);
+
+                if (!$file->exists() || !$file->isImage) {
+                    continue;
+                }
+
+                $images[$fileModel->path] = static::prepareImage($fileModel, $options);
+            }
+            // Folders
+            else {
+                $subFileModels = \FilesModel::findByPid($fileModel->uuid);
+
+                if (null === $subFileModels) {
+                    continue;
+                }
+
+                foreach ($subFileModels as $subFileModel) {
+                    if ('file' === $subFileModel->type) {
+                        $file = new \File($subFileModel->path, true);
+
+                        if (!$file->exists() || !$file->isImage) {
+                            continue;
+                        }
+
+                        $images[$subFileModel->path] = static::prepareImage($subFileModel, $options);
```

Comment:
Yeah did that on purpose because it's not recursive anyway and as the methods are static I felt like it's not worth putting it into a static private method.